server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIteratorForSingleChunk.java

```diff
@@ -0,0 +1,257 @@
+    "i.severity",
+
+    // column 11
+    "i.manual_severity",
+    "i.checksum",
+    "i.status",
+    "i.effort",
+    "i.author_login",
+    "i.issue_close_date",
+    "i.issue_creation_date",
+    "i.issue_update_date",
+    "r.plugin_name",
+    "r.plugin_rule_key",
+
+    // column 21
+    "r.language",
+    "p.uuid",
+    "p.module_uuid_path",
+    "p.path",
+    "p.scope",
+    "i.tags",
+    "i.issue_type"
+  };
+
+  private static final String SQL_ALL = "select " + StringUtils.join(FIELDS, ",") + " from issues i " +
+    "inner join rules r on r.id=i.rule_id " +
+    "inner join projects p on p.uuid=i.component_uuid " +
+    "inner join projects root on root.uuid=i.project_uuid";
+
+  private static final String PROJECT_FILTER = " AND root.uuid=?";
+  private static final String ISSUE_KEY_FILTER_PREFIX = " AND i.kee IN (";
+  private static final String ISSUE_KEY_FILTER_SUFFIX = ")";
+
+  static final Splitter TAGS_SPLITTER = Splitter.on(',').trimResults().omitEmptyStrings();
+  static final Splitter MODULE_PATH_SPLITTER = Splitter.on('.').trimResults().omitEmptyStrings();
+
+  private final DbSession session;
+
+  @CheckForNull
+  private final String projectUuid;
+
+  @CheckForNull
+  private final Collection<String> issueKeys;
+
+  private final PreparedStatement stmt;
+  private final ResultSetIterator<IssueDoc> iterator;
+
+  IssueIteratorForSingleChunk(DbClient dbClient, @Nullable String projectUuid, @Nullable Collection<String> issueKeys) {
+    this.projectUuid = projectUuid;
+    this.issueKeys = issueKeys;
+    this.session = dbClient.openSession(false);
+    try {
+
+      String sql = createSql();
+      stmt = dbClient.getMyBatis().newScrollingSelectStatement(session, sql);
+      try {
+        setParameters(stmt);
+        iterator = new IssueIteratorInternal(stmt);
+      } catch (SQLException e) {
+        DatabaseUtils.closeQuietly(stmt);
+        throw new IllegalStateException("Fail to prepare SQL request to select all issues", e);
+      }
+    } catch (Exception e) {
+      session.close();
+      throw new IllegalStateException("Fail to prepare SQL request to select all issues", e);
+    }
+  }
+
+  @Override
+  public boolean hasNext() {
+    return iterator.hasNext();
+  }
+
+  @Override
+  public IssueDoc next() {
+    return iterator.next();
+  }
+
+  private String createSql() {
+    String sql = SQL_ALL;
+    sql += projectUuid == null ? "" : PROJECT_FILTER;
+    if (issueKeys != null && !issueKeys.isEmpty()) {
+      sql += ISSUE_KEY_FILTER_PREFIX;
+      sql += IntStream.range(0, issueKeys.size()).mapToObj(i -> "?").collect(Collectors.joining(","));
+      sql += ISSUE_KEY_FILTER_SUFFIX;
+    }
+    return sql;
+  }
+
+  private void setParameters(PreparedStatement stmt) throws SQLException {
+    int index = 1;
+    if (projectUuid != null) {
+      stmt.setString(index++, projectUuid);
+    }
+    if (issueKeys != null) {
+      for (String key : issueKeys) {
+        stmt.setString(index++, key);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Extract this increment or decrement operator into a dedicated statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS881)